pkg/server/plugin/bundlepublisher/awss3/awss3.go

```diff
@@ -70,6 +72,11 @@ func buildConfig(coreConfig catalog.CoreConfig, hclText string, status *pluginco
 	if newConfig.Format == "" {
 		status.ReportError("configuration is missing the bundle format")
 	}
+	if newConfig.Endpoint != "" {
+		if _, err := url.ParseRequestURI(newConfig.Endpoint); err != nil {
+			status.ReportError("bad endpoint url format")
```

Comment:
Maybe something like this to also include the error:
```suggestion
			status.ReportErrorf("could not parse endpoint url: %v", err)
```